### PR TITLE
path-matchers

### PR DESCRIPTION
**Pull request by lifeart**
@knownasilya could you check pods logic?

### CODE REVIEW

File: test/utils/path-matcher-test.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+describe('PodMatcher', () => {
+  const matcher = new PodMatcher();
+  function m(str: string) {
+    return matcher.metaFromPath(str);
+  }
+  it('components', () => {
+    expect(m('foo/bar/app/pods/foo/component.ts')).toEqual({ type: 'component', name: 'foo' });
+    expect(m('foo/bar/app/pods/foo/component.js')).toEqual({ type: 'component', name: 'foo' });
+    expect(m('foo/bar/app/pods/foo/component.hbs')).toEqual({ type: 'component', name: 'foo' });
+    expect(m('foo/bar/app/pods/foo/component.hbs')).toEqual({ type: 'component', name: 'foo' });
```
Comment (knownasilya) [inline test/utils/path-matcher-test.ts]: Dupe check?

**Review by knownasilya — APPROVED** [review]
Looks good. Just wondering if this supports a mixed scenario, for example: `app/models/user.js` and `app/pods/myroute/route.js` exist in the same app.

File: test/utils/path-matcher-test.ts (hunk 1 above)
Comment (lifeart) [reply test/utils/path-matcher-test.ts]: Nice catch!

**Comment by lifeart** [comment]
> Looks good. Just wondering if this supports a mixed scenario, for example: `app/models/user.js` and `app/pods/myroute/route.js` exist in the same app.

yes, it will do double lookup -> classc, pods, if classic found - it wins

**Pull request merged**